resources/prosody-plugins/mod_token_verification.lua

```diff
@@ -70,3 +77,30 @@ module:hook("muc-occupant-pre-join", function(event)
 	log("debug", "pre join: %s %s", tostring(room), tostring(stanza));
 	return verify_user(origin, stanza);
 end);
+
+for event_name, method in pairs {
+    -- Normal room interactions
+    ["iq-set/bare/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+    -- Host room
+    ["iq-set/host/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+} do
+    module:hook(event_name, function (event)
+        local session, stanza = event.origin, event.stanza;
+
+        -- if disable room mod is false we pass it through, default behaviour
+        -- or the request is coming from admin (focus)
+        if not disable_room_moderation_on_empty_token or is_admin(stanza.attr.from) then
```

Comment:
This is hard to reason about (triple negation: not, disable, "empty_token"). If it was `if not require_token_for_moderation ...` it wouldn't even need a comment